### PR TITLE
refactor: isolate V1 ingester->querier client

### PR DESCRIPTION
**Pull request by crepererum**
Isolate the actual client from the query planning parts (`Ingester{Chunk,Partition}`) so we can hook up the V2 client in #8350.

The PR looks large, but it just moves code around and decouples the error handling.

### CODE REVIEW

**Comment by crepererum** [comment]
upsi, need to bring back some commented-out test code :sweat_smile:

File: querier/src/ingester/v1/mod.rs [hunk 1]
```diff
@@ -0,0 +1,1714 @@
...
+                None => panic!("Column not found: {}", desired_field.name()),
+            }
+        })
+        .collect::<Result<Vec<_>>>()?;
+
+    RecordBatch::try_new(expected_schema.as_arrow(), new_columns).context(CreatingRecordBatchSnafu)
+}
+
+// #[cfg(test)]
+// mod tests {
```
Comment (mhilton) [inline querier/src/ingester/v1/mod.rs:849]: Did you intend to keep all these tests commented out? If so will there be a follow-up to reenable them?

**Review by mhilton — COMMENTED** [review]
I see this is a draft so my comment will probably be out-of-date. But I reviewed it anyway.

File: querier/src/ingester/v1/mod.rs (hunk 1 above)
Comment (crepererum) [reply querier/src/ingester/v1/mod.rs]: fixed

**Review by mhilton — APPROVED** [review]
I didn't go through and check that all the code was moved from somewhere else, I'll take your word for that. The code that is there seems fine though.

**Pull request merged**